[32mINFO    [0m integration.high_availability.conftest:conftest.py:29 Clearing continuous writes
[32mINFO    [0m integration.high_availability.conftest:conftest.py:32 Starting continuous writes
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:127 Get leader unit
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:132 Run pre-upgrade-check action
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:138 Inject dependency fault
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:143 Refresh the charm
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:146 Wait for upgrade to fail on leader
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:152 Ensure continuous_writes while in failure state
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:343 Ensure continuous writes are incrementing
[32mINFO    [0m integration.helpers:helpers.py:485 Retrieving primary unit
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:368 max_written_value=377 on unit mysql/0
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:368 max_written_value=377 on unit mysql/0
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:368 max_written_value=494 on unit mysql/0
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:368 max_written_value=495 on unit mysql/1
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:368 max_written_value=498 on unit mysql/2
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:155 Re-run pre-upgrade-check action
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:158 Re-refresh the charm
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:160 Wait for upgrade to start
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  mysql/0 [idle] blocked: ready to rollback application
  mysql/1 [idle] waiting: other units upgrading first...
  mysql/2 [idle] waiting: other units upgrading first...
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  mysql/0 [idle] waiting: other units upgrading first...
  mysql/1 [idle] waiting: other units upgrading first...
  mysql/2 [executing] maintenance: starting services...
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  mysql/0 [executing] waiting: other units upgrading first...
  mysql/1 [idle] maintenance: upgrade completed
  mysql/2 [executing] maintenance: upgrade completed
[32mINFO    [0m juju.model:model.py:2972 Waiting for model:
  mysql/0 [idle] active: 
  mysql/1 [idle] active: 
  mysql/2 [idle] active: Primary
[32mINFO    [0m integration.high_availability.test_upgrade:test_upgrade.py:167 Ensure continuous_writes after rollback procedure
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:343 Ensure continuous writes are incrementing
[32mINFO    [0m integration.helpers:helpers.py:485 Retrieving primary unit
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:368 max_written_value=6264 on unit mysql/0
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:368 max_written_value=6264 on unit mysql/0
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:368 max_written_value=6320 on unit mysql/0
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:368 max_written_value=6325 on unit mysql/1
[32mINFO    [0m integration.high_availability.high_availability_helpers:high_availability_helpers.py:368 max_written_value=6327 on unit mysql/2
[32mINFO    [0m integration.high_availability.conftest:conftest.py:37 Clearing continuous writes
[32mINFO    [0m pytest_operator.plugin:plugin.py:790 Model status:

Model  Controller           Cloud/Region         Version  SLA          Timestamp
test   localhost-localhost  localhost/localhost  3.4.5    unsupported  01:46:50Z

App             Version          Status  Scale  Charm           Channel      Rev  Exposed  Message
mysql           8.0.37-0ubun...  active      3  mysql                          2  no       
mysql-test-app  0.0.2            active      1  mysql-test-app  latest/edge   63  no       

Unit               Workload  Agent  Machine  Public address  Ports           Message
mysql-test-app/0*  active    idle   0        10.124.253.215                  
mysql/0*           active    idle   1        10.124.253.33   3306,33060/tcp  
mysql/1            active    idle   2        10.124.253.66   3306,33060/tcp  
mysql/2            active    idle   3        10.124.253.152                  Primary

Machine  State    Address         Inst id        Base          AZ  Message
0        started  10.124.253.215  juju-7965d7-0  ubuntu@22.04      Running
1        started  10.124.253.33   juju-7965d7-1  ubuntu@22.04      Running
2        started  10.124.253.66   juju-7965d7-2  ubuntu@22.04      Running
3        started  10.124.253.152  juju-7965d7-3  ubuntu@22.04      Running

[32mINFO    [0m pytest_operator.plugin:plugin.py:796 Juju error logs:

machine-0: 01:35:05 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-0: 01:35:05 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-1: 01:35:06 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-1: 01:35:06 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-mysql-test-app-0: 01:35:06 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-mysql-0: 01:35:07 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-2: 01:35:14 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-2: 01:35:14 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-3: 01:35:14 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-3: 01:35:14 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-mysql-1: 01:35:14 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-mysql-2: 01:35:14 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-mysql-0: 01:43:38 ERROR unit.mysql/0.juju-log charms.data_platform_libs.v0.upgrade.VersionError - {'message': 'Versions incompatible, charm 1 can not be upgraded to 2', 'cause': 'Upgrades only supported for specific versions, charm versions satisfying requirement >1', 'resolution': ''}
unit-mysql-0: 01:43:38 CRITICAL unit.mysql/0.juju-log Upgrade failed, follow the instructions below to rollback:
    1. Re-run `pre-upgrade-check` action on the leader unit to enter 'recovery' state
    2. Run `juju refresh` to the previously deployed charm revision or local charm file

[32mINFO    [0m pytest_operator.plugin:plugin.py:862 Forgetting main...